src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java

```diff
@@ -0,0 +1,54 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.legacy.drawer.panel.DrawPanelController;
+import edu.kis.legacy.drawer.shape.ILine;
+import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
+
+
+public class LineDrawerAdapter implements Job2dDriver {
+    private int startX = 0, startY = 0;
+    private DrawPanelController drawPanelController;
+    private ILine line;
+
+    public LineDrawerAdapter(String lineType) {
+        super();
+        this.drawPanelController = DrawerFeature.getDrawerController();
+        lineType = lineType.toLowerCase();
+
+        switch (lineType) {
+            case "dotted":
+                this.line = LineFactory.getDottedLine();
+                break;
+            case "special":
+                this.line = LineFactory.getSpecialLine();
+                break;
+            default:
+                this.line = LineFactory.getBasicLine();
```

Comment:
Unfortunatelly, this is not a generic solution. (e.g. using other line types is not possible, unnecessary direct coupling)